C++/include/algorithm/dynamic_programming/coin_change.hpp

```diff
@@ -0,0 +1,65 @@
+/*
+    Coin Change Algorithm
+    ------------------------------------
+    Given an unlimited amount of coins with different values,
+    find the number of ways of making change for an given value using the coins.
+    This is a essential algorithm in dynamic programming that explores important
+    concepts.
+
+    Time complexity
+    ----------------
+    O(M*N), where M is the number of coins with different values and N is the amount that we desired to change
+
+    Space complexity
+    ----------------
+    O(M*N), where M is the number of coins with different values and N is the amount that we desired to change
+*/
+
+#ifndef COIN_CHANGE_HPP
+#define COIN_CHANGE_HPP
+
+#include <vector>
+using namespace std;
+
+int coin_change(const vector<int> coin, int number_of_coins, int amount)
+{
+    int i, j;
+
+    //make a matrix of size (amount+1)*number_of_coins to tabulate the computed values
+    //table[i][j] represents no.of ways in which an 'i' can be made with just j type of coins available
+    int table[amount + 1][number_of_coins];

```

Comment:
This is a variable-length array and throws an error. Consider vectors instead, and see https://stackoverflow.com/questions/1887097/why-arent-variable-length-arrays-part-of-the-c-standard